viadot/task_utils.py

```diff
@@ -527,3 +530,41 @@ def git_clone_url(self):
         if self.use_ssh:
             return f"git@{self.repo_host}:{self.repo}"
         return f"https://{self.git_token_secret}@{self.repo_host}/{self.repo}"
+
+
+@task
+def credentials_loader(credentials_secret: str, vault_name: str = None) -> dict:
+    """
+    Function that gets credentials from azure Key Vault or PrefectSecret or from prefect local config.
```

Comment:
lat option `from prefect local config` is more like `local config` not prefect local config because you can use this code also locally.